### ISSUE TITLE
Resnet50 applying last pooling layer regardless pooling parameter

### ISSUE THREAD

**Issue by weishi258** [tensorflow/tensorflow#21437]
### Describe the problem
Resnet50.py apply pooling at last layer regardless pooling parameter.

### Source code / logs
Current:
  x = AveragePooling2D((7, 7), name='avg_pool')(x)
  if include_top:
    x = Flatten()(x)
    x = Dense(classes, activation='softmax', name='fc1000')(x)
  else:
    if pooling == 'avg':
      x = GlobalAveragePooling2D()(x)
    elif pooling == 'max':
      x = GlobalMaxPooling2D()(x)
Fix:
  if include_top:
    x = AveragePooling2D((7, 7), name='avg_pool')(x)
    x = Flatten()(x)
    x = Dense(classes, activation='softmax', name='fc1000')(x)
  else:
    if pooling == 'avg':
      x = GlobalAveragePooling2D()(x)
    elif pooling == 'max':
      x = GlobalMaxPooling2D()(x)
